### PR TITLE
Redesign documentation landing and subpages

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Applied redesign as per [mock-up](https://www.figma.com/file/KQSCmTqOjY4gfKvJPyvzj2/Canonical.com-latest-only?type=design&node-id=9999-32699&t=jovsgOJ31F3rH5s1-0)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: https://canonical-com-905.demos.haus/documentation
- Check each tab against design and see that it matches

## Issue / Card

Fixes

### CODE REVIEW

**Comment by mtruj013** [comment]
@lyubomir-popov I'm having some trouble recreating this. The [Vanilla tabs](https://vanillaframework.io/docs/patterns/tabs) don't quite behave this way, the line and the tabs start at the same point. Is there another pattern I should be using here? 
Screenshot from 2023-06-21 16-20-15

**Comment by ClementChaumel** [comment]
The dark bar on the left is a bit off
image

but I'm wondering @lyubomir-popov @juanruitina why is this not using the [table of content component](https://vanillaframework.io/docs/patterns/table-of-contents)? or at least the [Side navigation](https://vanillaframework.io/docs/patterns/navigation#side-navigation)

File: static/sass/styles.scss [hunk 1]
```diff
@@ -359,6 +358,27 @@ $p-small-lh-diff: map-get($line-heights, default-text) - map-get($line-heights,
...
+  a {
+    color: #3097ff;
+    padding-left: 0;
+
+    &:visited {
+      color: #AB83D3;
+    }
+  }
+ }
+}
```
Comment (ClementChaumel) [inline static/sass/styles.scss:374]: We should probably use a different pattern altogether but if we're going forward with this let's use vanilla colours!

File: static/sass/styles.scss [hunk 2]
```diff
@@ -1,5 +1,4 @@
-$color-tabs-active-bar: #e95420;
-
+$color-tabs-active-bar: #000;
```
Comment (ClementChaumel) [inline static/sass/styles.scss:1]: I don't think this is used?

File: static/sass/styles.scss [hunk 3]
```diff
@@ -359,6 +358,27 @@ $p-small-lh-diff: map-get($line-heights, default-text) - map-get($line-heights,
...
+ ul li {
+  background: #f3f3f3;
+
+  a {
+    color: #3097ff;
+    padding-left: 0;
+
+    &:visited {
+      color: #AB83D3;
+    }
```
Comment (ClementChaumel) [inline static/sass/styles.scss:371]:
```suggestion
  background: $color-paper;

  a {
    color: $color-link;
    padding-left: 0;

    &:visited {
      color: $color-link-visited;
    }
```

**Review by ClementChaumel — APPROVED** [review]

**Comment by ClementChaumel** [comment]
> @lyubomir-popov I'm having some trouble recreating this. The [Vanilla tabs](https://vanillaframework.io/docs/patterns/tabs) don't quite behave this way, the line and the tabs start at the same point. Is there another pattern I should be using here? Screenshot from 2023-06-21 16-20-15

If we're going forward with this hr, we should push it 1px to the top 
image

(But I think the actual fix is to just use the vanilla tabs

**Comment by lyubomir-popov** [comment]
> but I'm wondering @lyubomir-popov @juanruitina why is this not using the [table of content component](https://vanillaframework.io/docs/patterns/table-of-contents)? or at least the [Side navigation](https://vanillaframework.io/docs/patterns/navigation#side-navigation)

I had used a list of anchors in the design, because this isn't a sidenav - but let's just use the default sidenav, seems like less effort from where things curently are

**Comment by ClementChaumel** [comment]
> nav - but let's just use the default sidenav, seems like less effort from where things curently are

In function, it is a table of content. Why are we not using the existing pattern, are we deprecating the vanilla component?

File: static/sass/styles.scss (hunk 2 above)
Comment (mtruj013) [reply static/sass/styles.scss:1]: It is, it's the bottom border color for active tabs here: 
Screenshot from 2023-06-22 11-46-38

with the old color:
Screenshot from 2023-06-22 11-46-47

**Comment by lyubomir-popov** [comment]
> > nav - but let's just use the default sidenav, seems like less effort from where things curently are
> 
> In function, it is a table of content. Why are we not using the existing pattern, are we deprecating the vanilla component?

You're right. Let's just use that. I needs improving but I can do that as a vanilla pr and it will then update here. I'll put it on my list.

**Comment by juanruitina** [comment]
@lyubomir-popov We already have this open: https://github.com/canonical/vanilla-framework/issues/4777 Let's work together on it.

**Comment by lyubomir-popov** [comment]
@mtruj013 should we have breadcrumbs here? cc @juanruitina

**Comment by juanruitina** [comment]
I don't think we need breadcrumbs here, they would be too redundant with the tabs.

**Comment by ClementChaumel** [comment]
I'm switching the target to the refresh branch. Undo if it's not correct

**Comment by lyubomir-popov** [comment]
Should we close this one @mtruj013? I think we agreed to continue work on the other one

**Pull request closed without merging**